packages/components/app/styles/components/table.scss

```diff
@@ -19,175 +19,170 @@ $hds-table-cell-padding-medium: 14px 16px 13px 16px; // the 1px difference is to
 $hds-table-cell-padding-short: 6px 16px 5px 16px; // the 1px difference is to account for the bottom border
 $hds-table-cell-padding-tall: 22px 16px 21px 16px; // the 1px difference is to account for the bottom border
 
+
+// TABLE
+
 .hds-table {
   width: 100%;
   border: $hds-table-border-width solid $hds-table-border-color;
   border-radius: $hds-table-border-radius;
   border-spacing: 0;
 }
 
-// TABLE DISPLAY
+
+// table display
+
 .hds-table--layout-fixed {
   table-layout: fixed;
 }
 
+
+// ----------------------------------------------------------------
+
 // TABLE HEADER
+
 .hds-table__thead {
   .hds-table__tr {
     color: var(--token-color-foreground-strong);
     background-color: var(--token-color-surface-strong);
-    // BORDER RADIUS: TARGET FIRST AND LAST TH ELEMENTS IN THE ROW
-    &:first-of-type {
-      th:first-child {
-        border-top-left-radius: $hds-table-inner-border-radius;
+
+    th {
+      position: relative;
+      padding: $hds-table-cell-padding-medium;
+      text-align: left;
+      border-top: none;
+      border-right: none;
+      border-bottom: $hds-table-border-width solid $hds-table-border-color;
+      border-left: none;
+
+      // right border (we emulate a right border slightly detached from the top/bottom borders)
+
+      &::after {
+        position: absolute;
+        top: 6px;
+        right: 0;
+        bottom: 6px;
+        width: 1px;
+        background-color: $hds-table-border-color;
+        content: "";
+        pointer-events: none;
       }
```

Comment:
This is fine, but this is how I prefer to handle this kind of "in-between" item in order to avoid needing an additional `:last-child` style to override it.

```
+ th::before {
    position: absolute;
    top: 6px;
    left: -1px; // positions it where the actual th border would be if "turned on"
    bottom: 6px;
    width: 1px;
    background-color: var(--token-color-border-primary);
    content: "";
    pointer-events: none;
}
```

(Final selector in browser would be something like: `.hds-table__thead .hds-table__tr th + th::before`)

Reply:
@KristinLBradley keep in mind that I've done a large refactoring of this Sass code (see https://github.com/hashicorp/design-system/pull/1860/commits/46526e38a90b00ff2d9f62b7f297012aaf1dce9f + https://github.com/hashicorp/design-system/pull/1860/commits/1ae7f227f7551a6d270d21dfa63327324c7a8c51) so I would not focus too much on the layout/styling details (until I've done the rebase of this branch on the other one) and more on the JavaScript for [select/unselect] [all/row]